blueprints/validations.py

```diff
@@ -0,0 +1,49 @@
+"""Module for validation actions inside of Blueprints."""
+
+
+class NonPositiveValueError(Exception):
+    """Raised when a value is less than or equal to zero."""
+
+
+class NegativeValueError(Exception):
+    """Raised when a value is negative."""
+
+
+def raise_if_less_or_equal_to_zero(**kwargs) -> None:
```

Comment:
question (not specifically about this PR): do we want to force kwargs input for our functions/classes? I'm not sure if our current signature are already in this style: "Formula(self,*,a,b,c). But that would do the trick.

Reply:
That's a good point. We can make it more general by requesting a dictionary, rather than enforcing the use of kwargs entirely. The place where this is implemented should then unpack the kwargs first.

```python
def raise_if_negative(values: dict[str, float]) -> None:
    """
    Raise a NegativeValueError if any value in the provided dictionary is negative.

    Parameters
    ----------
    values : dict[str, float]
        A dictionary where keys are parameter names, and values are the values to validate.

    Raises
    ------
    NegativeValueError
        If any value in the dictionary is negative.
    """
    # Iterate over each item in the dictionary
    for key, value in values.items():
        # Check if the value is negative
        if value < 0:
            # If a negative value is found, raise the NegativeValueError with a message
            raise NegativeValueError(f"Invalid value for '{key}': {value}. Values for '{key}' cannot be negative.")
```

Reply:
Mwaah that's not my point, my point is more on the init side ;-)

I think this is more-or-less the same as using **kwargs, the usage is a bit different ofcourse:

`self.raise_if_negative(a=1,b=2,c=3)`

vs 

`self.raise_if_negative({"a":1,"b":2,"c":3})`


I think I like the current implementation (first) better...